nexus/db-queries/src/db/datastore/volume_repair.rs

```diff
@@ -39,15 +40,19 @@ impl DataStore {
             .execute_async(&*conn)
             .await
             .map(|_| ())
-            .map_err(|e| {
-                if e.to_string().contains(
-                    "duplicate key value violates \
-                    unique constraint \"volume_repair_pkey\"",
-                ) {
-                    Error::conflict("volume repair lock")
-                } else {
-                    public_error_from_diesel(e, ErrorHandler::Server)
-                }
+            .map_err(|e| match e {
+                DieselError::DatabaseError(
+                    DatabaseErrorKind::UniqueViolation,
+                    ref error_information,
+                ) => match error_information.constraint_name() {
+                    Some("volume_repair_pkey") => {
+                        Error::conflict("volume repair lock")
+                    }
+
+                    _ => public_error_from_diesel(e, ErrorHandler::Server),
+                },
+
+                _ => public_error_from_diesel(e, ErrorHandler::Server),
```

Comment:
nit, take it or leave it: I think this could be simplified to
```suggestion
                DieselError::DatabaseError(
                    DatabaseErrorKind::UniqueViolation,
                    ref error_information,
                ) if error_information.constraint_name() == Some("volume_repair_pkey") => {
                    Error::conflict("volume repair lock")
                },
                _ => public_error_from_diesel(e, ErrorHandler::Server),
```

Reply:
thanks! done in 8d1bf6714